kernel/src/log_segment.rs

```diff
@@ -139,6 +139,17 @@ impl LogSegment {
     /// [`Snapshot`]: crate::snapshot::Snapshot
     #[internal_api]
     pub(crate) fn for_snapshot(
+        storage: &dyn StorageHandler,
+        log_root: Url,
+        time_travel_version: impl Into<Option<Version>>,
+    ) -> DeltaResult<Self> {
+        let time_travel_version = time_travel_version.into();
```

Comment:
is this needed? can we not just pass it along as is?

Reply:
oh or if you mean making it `Option<Version>` - that seems reasonable.. and just push the `time_travel_version.into()` above? - tbh I feel like we don't really need the `Into` much at all but maybe won't change that here